File: JobApplicationMicroservice/src/main/java/com/licenta/jobapplicationmicroservice/business/interfaces/IDatabaseService.java
package com.licenta.jobapplicationmicroservice.business.interfaces;

import com.licenta.jobapplicationmicroservice.business.model.Company;
import com.licenta.jobapplicationmicroservice.business.model.Job;

public interface IDatabaseService {

    void verifyUserExists(Long userId);
    Job getJob(Long jobId);
    Company getCompany(Long companyId);
}

File: NewsfeedMicroservice/src/main/kotlin/com/licenta/newsfeedmicroservice/business/interfaces/IDatabaseService.kt
package com.licenta.newsfeedmicroservice.business.interfaces

import com.licenta.newsfeedmicroservice.business.model.*

interface IDatabaseService {

    fun getCompanyIndustriesFollowedByUser(userId: Long): List<CompanyIndustry>
    fun getCityById(cityId: Int): City
    fun getCompanyById(companyId: Long): Company
    fun getCountryById(countryId: Int): Country
    fun getWorkTypeById(workTypeId: Int): WorkType
}